packages/react-server/src/ReactFizzServer.js

```diff
@@ -2232,30 +2250,18 @@ function renderNodeDestructiveImpl(
             'Render them conditionally so that they only appear on the client render.',
         );
       case REACT_LAZY_TYPE: {
+        const previousComponentStack = task.componentStack;
+        task.componentStack = getBuiltInComponentStack(task, 'Lazy');
         const lazyNode: LazyComponentType<any, any> = (node: any);
         const payload = lazyNode._payload;
         const init = lazyNode._init;
-        let resolvedNode;
-        if (__DEV__) {
-          try {
-            resolvedNode = init(payload);
-          } catch (x) {
-            if (
-              typeof x === 'object' &&
-              x !== null &&
-              typeof x.then === 'function'
-            ) {
-              // this Lazy initializer is suspending. push a temporary frame onto the stack so it can be
-              // popped off in spawnNewSuspendedTask. This aligns stack behavior between Lazy in element position
-              // vs Component position. We do not want the frame for Errors so we exclusively do this in
-              // the wakeable branch
-              pushBuiltInComponentStackInDEV(task, 'Lazy');
-            }
-            throw x;
```

Comment:
I was originally motivated to remove this special case because I didn't want to have the extra try in prod. However I'm also not sure this was correct to begin with. The comment (which I wrote) indicates we are trying to have parity between stack behavior with element and Component position lazy's. However When I test the client behavior there is a seemingly buggy thing today where as an element if the lazy itself rejects you are told there is an error in <Offscreen /> component. However by removing this special case now elements and components both get a `Lazy` frame if the promise rejects and if the promise resolves this frame is removed from any future errors that arise deeper in the render. I think this makes sense but I wrote this comment for a reason and I no longer remember why. For now I'm going with this until we can clarify what the proper client behavior is since it should match